service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -424,19 +497,37 @@ WHERE employee_id = :userId
         throw Forbidden("Permission denied")
     }
 
+    private inline fun <reified A> hasPermissionThroughRoles(
```

Comment:
I'm not 100% sure, but couldn't all these functions be normal non-inline non-reified functions? Usually reified type parameters are used when there's some need to operate on the type directly (e.g. in this case calling `A::class` would require a reified function). But it looks like we're just doing simple checks based on values and functions here